actors/miner/src/lib.rs

```diff
@@ -1950,6 +1951,238 @@ impl Actor {
+            &pledge_inputs.network_baseline,
+            &pledge_inputs.epoch_reward,
+            &pledge_inputs.network_qap,
+            &pledge_inputs.circulating_supply,
+        );
+
+        let day_reward = expected_reward_for_power(
+            &pledge_inputs.epoch_reward,
+            &pledge_inputs.network_qap,
+            &base_sector_power,
+            fil_actors_runtime::EPOCHS_IN_DAY,
+        );
+
+        let storage_pledge = expected_reward_for_power(
+            &pledge_inputs.epoch_reward,
+            &pledge_inputs.network_qap,
+            &base_sector_power,
+            INITIAL_PLEDGE_PROJECTION_PERIOD,
+        );
+
+        let sector_verify_info = params
+            .sectors
+            .iter()
+            .zip(params.sector_proofs.into_iter())
+            .map(|(sector, proof)| SealVerifyInfo {
+                registered_proof: params.seal_proof_type,
+                sector_id: fvm_shared::sector::SectorID {
+                    miner: miner_id,
+                    number: sector.sector_number,
+                },
+                randomness: Randomness(Vec::new()),
+                proof: proof.into(),
+                sealed_cid: sector.sealed_cid,
+                unsealed_cid: CompactCommD::empty().get_cid(params.seal_proof_type).unwrap(),
+                deal_ids: vec![],
+                interactive_randomness: Randomness(Vec::new()),
+            })
+            .collect::<Vec<_>>();
+
+        let sector_verifications = rt
+            .batch_verify_seals(&sector_verify_info)
+            .context_code(ExitCode::USR_ILLEGAL_ARGUMENT, "failed to batch verify")?;
+
+        let sectors_to_add = params
+            .sectors
+            .into_iter()
+            .zip(sector_verifications)
+            .filter(|(_, verified)| *verified)
+            .map(|(sector, _)| SectorOnChainInfo {
+                sector_number: sector.sector_number,
+                seal_proof: params.seal_proof_type,
+                sealed_cid: sector.sealed_cid,
+                deprecated_deal_ids: vec![],
+                expiration: sector.expiration,
+                activation: activation_epoch,
+                deal_weight: DealWeight::zero(),
+                verified_deal_weight: DealWeight::zero(),
+                initial_pledge: initial_pledge.clone(),
+                expected_day_reward: day_reward.clone(),
+                expected_storage_pledge: storage_pledge.clone(),
+                power_base_epoch: activation_epoch,
+                replaced_day_reward: TokenAmount::zero(),
+                sector_key_cid: None,
+                flags: SectorOnChainInfoFlags::SIMPLE_QA_POWER,
+            })
+            .collect::<Vec<SectorOnChainInfo>>();
+
+        let total_pledge = BigInt::from(sectors_to_add.len()) * initial_pledge;
+
+        rt.transaction(|state: &mut State, rt| {
+            let current_balance = rt.current_balance();
+            if current_balance < total_pledge {
```

Comment:
You need to check the _available_ balance. Use `state.get_unlocked_balance(&current_balance)`